### ISSUE TITLE
formating bug

### ISSUE THREAD

**Issue by rajneeshojha** [beautify-web/js-beautify#845]
@if($gateway ) <div class="row"> <div class="col-xs-12"> </div> </div> @endif

formating this piece of code which include blade template, removes first line " @if($gateway )"

**Comment by bitwiseman** [comment]
We don't support that template language at this time.  Where can we get info on that syntax?  It is unlikely to be implemented in the near term, but it would be good to have details.

**Comment by rajneeshojha** [comment]
It is a blade template

**Comment by bitwiseman** [comment]
You mean this? 
https://laravel.com/docs/5.1/blade

**Comment by nivv** [comment]
@bitwiseman yes, that's correct. In my opinion it would be a good start to be able to ignore `@`-tags completely

**Comment by whereisjefe** [comment]
as @nivv said, just ignoring the @ tag would be sufficient to stop the issues i have with blades

**Comment by stevendesu** [comment]
It's not a perfect solution, but since little headway has been made in the last year (and no other HTML formatter that I found properly handled blade templates) I created the following:

https://github.com/stevendesu/html-formatter

It's still a long-shot from "production ready", but hopefully it will help some people out. I'm also accepting all pull requests and assistance.

**Comment by saeedphr** [comment]
any solution ?

**Comment by stevendesu** [comment]
@saeedphr GitHub parses HTML, so I think the HTML comment got stripped in your comment there 😆 

You might want to put the solution in a code block

    like so

**Comment by bitwiseman** [comment]
@stevendesu - You started a whole new project?  I realize there's less code, but if you're going to go that far, come help with this one.  😄

**Comment by stevendesu** [comment]
@bitwiseman At the time I looked at creating a PR for this project, or two other similar projects (projects which format HTML)

Unfortunately most of the projects were whole lexers and token analyzers and were complicated and would have taken several hours to understand before I ever made any additions to the code.

It was just quicker to import a DOM parser and re-build the HTML tree with the desired formatting 😄 Which is what I did

**Comment by brnmonteiro** [comment]
If helps: (little hack)[https://gist.github.com/brnmonteiro/3660b71dbc68691cdc8ac41cec379e2f]
isn't the best approach, but works...

**Comment by nivv** [comment]
@brnmonteiro it does not work.

Does not work with `@if`-statements or `@foreach`.

```blade
// Before
@section('content')
    @if($page == 1)
        {!! $page !!}
    @endif
@endsection
```

```blade
// After
@section('content')
@if($page == 1) {!! $page !!}
@endif
@endsection
```

**Comment by bitwiseman** [comment]
@FaridAghili suggested this further hack. 
https://gist.github.com/mpryvkin/0c46e2493b450f92492e8e9a46ad5d97

Seems based on @brnmonteiro 's hack.  If there were sufficient tests to show this worked, we'd consider accepting it.  I'm thinking there'd need to be quite a number of tests though.

**Comment by shamilovtim** [comment]
It would be nice to merge this. Since VSCode supposedly uses js-beautify internally, blade templating is broken with the current indentation system and all blade directives and child HTML need to be indented manually. This is quite frustrating and time consuming.

If there's any way I can contribute, please let me know.

**Comment by shamilovtim** [comment]
There are bugs in @FaridAghili's version which I've raised in the gist. Haven't tried the initial gist this is about yet.
